### ISSUE TITLE
False positive with compare 

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#577]
# Bug report

## What's wrong

```python
assert await _coroutine(1) == Success(1)
```

Raise incorrect order violation:

```
tests/test_functions/test_pipeline.py:61:12: Z309 Found reversed comparison order
```

**Comment by sobolevn** [comment]
Closed with `0.9.0`
